src/servers/src/postgres/types/datetime.rs

```diff
@@ -0,0 +1,406 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use bytes::BufMut;
+use chrono::{NaiveDate, NaiveDateTime};
+use pgwire::types::ToSqlText;
+use postgres_types::{IsNull, ToSql, Type};
+use session::session_config::{PGDateOrder, PGDateTimeStyle};
+
+#[derive(Debug)]
+pub struct StylingDate<'a>(pub &'a NaiveDate, pub PGDateTimeStyle, pub PGDateOrder);
+
+#[derive(Debug)]
+pub struct StylingDateTime<'a>(pub &'a NaiveDateTime, pub PGDateTimeStyle, pub PGDateOrder);
+
+fn date_format_string(style: PGDateTimeStyle, order: PGDateOrder) -> &'static str {
+    match style {
+        PGDateTimeStyle::ISO => "%Y-%m-%d",
+        PGDateTimeStyle::German => "%d.%m.%Y",
+        PGDateTimeStyle::Postgres => match order {
+            PGDateOrder::MDY | PGDateOrder::YMD => "%m-%d-%Y",
+            PGDateOrder::DMY => "%d-%m-%Y",
+        },
+        PGDateTimeStyle::SQL => match order {
+            PGDateOrder::MDY | PGDateOrder::YMD => "%m/%d/%Y",
+            PGDateOrder::DMY => "%d/%m/%Y",
+        },
+    }
+}
```

Comment:
Given that not all style has order support, we can reorganize this to

```rust
enum PGDateTimeStyle {
  ISO,
  German,
  Postgres(PGDateOrder),
  SQL(PGDateOrder),
}
```

Reply:
No, we can't do this. Because order and style can be set independently. For example, you can first set style to `ISO,MDY`, then set style to `Postgres`. If we don't trace the order, we will fail to know the order to use.

Reply:
Get